src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -197,109 +507,212 @@
+     * the static JVM-wide filter, or to create a filter from a pattern string.
+     * The static filter factory and the static filter apply to the whole Java runtime,
+     * or "JVM-wide", there is only one of each, for a complete description of
+     * the function and use refer to {@link ObjectInputFilter}.
+     *
+     * <p>The JVM-wide deserialization filter factory and the static JVM-wide filter
+     * can be configured from system properties during the initialization of the
+     * {@code ObjectInputFilter.Config} class.
+     *
+     * <p>If the Java virtual machine is started with the system property
      * {@systemProperty jdk.serialFilter}, its value is used to configure the filter.
      * If the system property is not defined, and the {@link java.security.Security} property
      * {@code jdk.serialFilter} is defined then it is used to configure the filter.
      * Otherwise, the filter is not configured during initialization and
      * can be set with {@link #setSerialFilter(ObjectInputFilter) Config.setSerialFilter}.
      * Setting the {@code jdk.serialFilter} with {@link System#setProperty(String, String)
      * System.setProperty} <em>does not set the filter</em>.
-     * The syntax for each property is the same as for the
+     * The syntax for the property value is the same as for the
      * {@link #createFilter(String) createFilter} method.
-     *
+     * <p>
+     * If the Java virtual machine is started with the system property
+     * {@systemProperty jdk.serialFilterFactory}, its value names the class to configure the
+     * JVM-wide deserialization filter factory.
+     * If the system property is not defined, and the {@link java.security.Security} property
+     * {@code jdk.serialFilterFactory} is defined then it is used to configure the filter factory.
+     * The class must be public, must have a public zero-argument constructor, implement the
+     * {@link BinaryOperator {@literal BinaryOperator<ObjectInputFilter>}} interface, provide its implementation and
+     * be accessible via the {@linkplain ClassLoader#getSystemClassLoader() application class loader}.
+     * The filter factory configured using the system or security property during initialization
+     * can NOT be replaced with {@link #setSerialFilterFactory(BinaryOperator) Config.setSerialFilterFactory}.
+     * This ensures that a filter factory set on the command line is not overridden accidentally
+     * or intentionally by the application.
+     * Setting the {@code jdk.serialFilterFactory} with {@link System#setProperty(String, String)
+     * System.setProperty} <em>does not set the filter factory</em>.
+     * The syntax for the system property value and security property value is the
+     * fully qualified class name of the deserialization filter factory.
      * @since 9
      */
     final class Config {
-        /* No instances. */
-        private Config() {}
+        /**
+         * Lock object for filter and filter factory.
```

Comment:
The lock is not used for the filter factory, is it?

Reply:
The lock was needed for the factory to make the check and setting atomic.